### PR TITLE
Updates for 2.4 release

### PR DESCRIPTION
**Pull request by bbayles**
We've got a batch of new and improved itertools, and the issue / PR queue is clear - :ship: !

This PR updates the changelog with the new items (used different formatting this time), indents the code samples in docstrings consistently (some were indented, some weren't), and bumps the version to 2.4.

@erikrose , if you're good with this and can release to PyPI, I will make the GitHub release (as with the last one).

### CODE REVIEW

File: more_itertools/more.py [hunk 1]
```diff
@@ -340,8 +340,18 @@ def one(iterable):
 def distinct_permutations(iterable):
```
Comment (bbayles) [inline more_itertools/more.py]: This one was missing an example. The `itertools` docs describe the number of elements returned for `permutations` and `combinations`, so I've added that here too.

**Comment by bbayles** [comment]
@erikrose, anything you think needs to be done before release? No rush from me, but we have some [friends](https://github.com/erikrose/more-itertools/issues/87) looking forward to 2.4.

**Comment by erikrose** [comment]
I don't have anything in mind. Let me give it a quick once-over, and I'll cut the release! Great work Bo and contribs again! It's wonderful to see the community take this project under its wing.

**Comment by erikrose** [comment]
I uploaded the new release. But upon trying to update the docs, I find that PyPI has abdicated its responsibility for holding docs:

> Upload failed (410): Uploading documentation is no longer supported, we recommend using https://readthedocs.org/.

Fun! So this is an excuse to move to the (tragically underfunded) readthedocs.

**Comment by erikrose** [comment]
I see you or I had already set up readthedocs, so that's cool. :-) I bumped to 2.4.1 and re-released so we could have the docs link at https://pypi.python.org/pypi/more-itertools/ point to the right place. (`python setup.py register`, an old trick I used to use to update the index page without re-releasing, is no longer supported by the server.)

**Comment by erikrose** [comment]
I haven't yet figured out how to remove the big Package Documentation link at the top that still points to pythonhosted.org, however. Hmm.

Btw, I'd like to make you a maintainer on the package so you can upload releases. What's your PyPI username?

**Comment by bbayles** [comment]
I don't recall setting up RTD, but I'm pleased to have things hosted there; RTD is great. (My [company](https://observable.net/) runs ads there in support of them and their ethical ad standards)

I am bbayles on PyPI as well.

**Comment by erikrose** [comment]
Oh good; I'm so glad they found a way to make money!

**Comment by pylang** [comment]
Thank you for this release.  It's awesome.

**Comment by bbayles** [comment]
[This thread](https://github.com/scrapy/cssselect/issues/65#issuecomment-263057732) describes how to redirect docs. I think there's a workaround, but it seems like it might be difficult to get right on the first try?

**Comment by bbayles** [comment]
Oh, actually there's a DESTROY DOCUMENTATION button now - it's on [this page](https://pypi.python.org/pypi?%3Aaction=pkg_edit&name=more-itertools). I tested the .zip file trick on another repo and that works too.

I'll go ahead and do the .zip file trick to redirect, since the old docs are easy to get on Google.

**Comment by erikrose** [comment]
Great. The redirect is superior anyway, in case anyone has bookmarked the page.

**Comment by erikrose** [comment]
I've wriggled through the rtd UI to "activate" some of the older versions, so they're now accessible via the menu in the lower left.

**Pull request merged**